plugins/civmodcore-paper/src/main/java/vg/civcraft/mc/civmodcore/config/ConfigHelper.java

```diff
@@ -269,14 +273,117 @@ public static ItemMap parseItemMapDirectly(@Nullable final ConfigurationSection
 				}
 
 			}
-			toAdd.setItemMeta(meta);
-			if (current.contains("nbt")) {
-				toAdd = ItemMap.enrichWithNBT(toAdd, 1, current.getConfigurationSection("nbt").getValues(true));
+			if (current.contains("max_stack_size")) {
+				meta.setMaxStackSize(current.getInt("max_stack_size"));
+			}
+			if (current.contains("damage")) {
+				((Damageable) meta).setDamage(current.getInt("damage"));
+			}
+			if (current.contains("rarity")) {
+				meta.setRarity(ItemRarity.valueOf(current.getString("rarity")));
+			}
+			if (current.contains("custom_model_data")) {
+				meta.setCustomModelData(current.getInt("custom_model_data"));
+			}
+			if (current.contains("hide_tooltip")) {
+				meta.setHideTooltip(current.getBoolean("hide_tooltip"));
+			}
+			if (current.contains("repair_cost")) {
+				net.minecraft.world.item.ItemStack nmsItem = ItemUtils.getNMSItemStack(toAdd);
+				nmsItem.set(DataComponents.REPAIR_COST, current.getInt("repair_cost"));
+
+				toAdd = nmsItem.getBukkitStack();
+			}
+			if (current.contains("enchantment_glint_override")) {
+				meta.setEnchantmentGlintOverride(current.getBoolean("enchantment_glint_override"));
+			}
+			if (current.contains("fire_resistant")) {
+				meta.setFireResistant(current.getBoolean("fire_resistant"));
+			}
+			if (current.contains("bundle_contents")) {
+				// TODO: doesn't look hard, just same logic as FM item config
 			}
+			if (current.contains("entity_data")) { // TODO: only doing ID for now but some other fields might be useful here
+				ConfigurationSection entityDataSection = current.getConfigurationSection("entity_data");
+				net.minecraft.world.item.ItemStack nmsItem = ItemUtils.getNMSItemStack(toAdd);
+
+				CustomData customData = CustomData.EMPTY.update(nbt -> nbt.putString("id", entityDataSection.getString("id")));
+				nmsItem.set(DataComponents.ENTITY_DATA, customData);
+				toAdd = nmsItem.getBukkitStack();
+			}
```

Comment:
Might be worth looking into serialised item stacks using `==:`, though we'd need to find/create an item creator that produces a YAML output.

Reply:
Yeah, I think that'd be a much better solution just because of the volume of stuff that items can have now that needs to be checked for and assigned. I put something together using Bukkit's inbuilt ItemStack serialization that works pretty well, as well as a command in SAH to quickly see what a serialized item should look like in a config.